modules/http4s/src/smithy4s/http4s/ClientEndpointMiddleware.scala

```diff
@@ -48,4 +48,18 @@ object ClientEndpointMiddleware {
       ): Client[F] => Client[F] = identity
     }
 
+  implicit def semigroupClientEndpointMiddleware[F[_]]
```

Comment:
**suggestion:** now that I think of it, we can do a full `Monoid`. `empty` would return a `ClientEndpointMiddleware` where `prepare` returns an identity function.

Same for servers.

Reply:
I'll implement the Monoid! Thank you

And sorry about the imports, I was a bit too eager and didn't get any diagnostics so I was like letsgoooo